en/services/mission.md

```diff
@@ -235,16 +239,89 @@ In more detail, the sequence of operations is:
+### Partial Mission Upload {#upload_partial}
 
-TBD
+The diagram below shows the communication sequence to upload a partial mission to a drone (assuming all operations succeed).
 
+> **Note** The message sequence is the same as a [full mission upload](#uploading_mission) except that it is triggered with a [MISSION_WRITE_PARTIAL_LIST](../messages/common.md#MISSION_WRITE_PARTIAL_LIST) instead of a [MISSION_COUNT](../messages/common.md#MISSION_COUNT).
+  An update is expected to be robust (succeed completely or fail and revert to original version).
 
-### Download Partial Mission {#download_partial}
+<span></span>
+> **Note** Unlike for full mission update, partial update is not required to be robust; individual commands can fail.
 
-TBD
+{% mermaid %}
+sequenceDiagram;
+    participant GCS
+    participant Drone
+    GCS->>Drone: MISSION_WRITE_PARTIAL_LIST
+    GCS->>GCS: Start timeout
+    Drone->>GCS: MISSION_REQUEST_INT (n)
+    GCS->>GCS: Start timeout
+    GCS-->>Drone: MISSION_ITEM_INT (n)
+    Drone->>GCS: MISSION_REQUEST_INT (n+1)
+    GCS->>GCS: Start timeout
+    GCS-->>Drone: MISSION_ITEM_INT (n+1)
+    Drone->>GCS: MISSION_ACK
+{% endmermaid %}
+
+In more detail, the sequence of operations is:
+1. GCS (client) sends [MISSION_WRITE_PARTIAL_LIST](../messages/common.md#MISSION_WRITE_PARTIAL_LIST) specifying the start and end index of items to overwrite.
+   - A [timeout](#timeout) must be started for the GCS to wait on the response from Drone (`MISSION_REQUEST_INT`).
+1. Drone (server) receives the message, and prepares to upload mission items.
+1. Drone responds with [MISSION_REQUEST_INT](../messages/common.md#MISSION_REQUEST_INT) requesting the first mission item (`seq==n`).
+1. GCS receives `MISSION_REQUEST_INT` and responds with the requested mission item in a [MISSION_ITEM_INT](../messages/common.md#MISSION_ITEM_INT) message.
+1. Drone receives [MISSION_ITEM_INT](../messages/common.md#MISSION_ITEM_INT) and writes it into the existing mission.
+1. Drone and GCS repeat the `MISSION_REQUEST_INT`/`MISSION_ITEM_INT` cycle, iterating `seq` until all items are uploaded.
+1. For the last mission item, the drone responds with [MISSION_ACK](../messages/common.md#MISSION_ACK) with the result of the operation result: `type` ([MAV_MISSION_RESULT](../messages/common.md#MAV_MISSION_RESULT)):
+   - On success, `type` must be set to [MAV_MISSION_ACCEPTED](../messages/common.md#MAV_MISSION_ACCEPTED)
+   - On failure, `type` must set to [MAV_MISSION_ERROR](../messages/common.md#MAV_MISSION_ERROR) or some other error code. 
+1. GCS receives `MISSION_ACK`:
+   - If `MAV_MISSION_ACCEPTED` the operation is complete.
+   - If an error, the transaction fails but may be retried. <!-- not clear here -->
+
+Note:
+- The GCS (client) sets a [timeout](#timeout) after every message and will resend if there is no response from the vehicle.
+- The client will re-request missing mission items if any are received out of sequence.
```

Comment:
Client earlier refereed to GCS, but the vehicle/remote who isn't uploading the mission is the only one who actually knows what items are missing.

Reply:
Yes. Thanks. Globally I'm going to stop referring to them as client and server. That model doesn't quite work.